### PR TITLE
Automate 10.x releases

### PR DESCRIPTION
**Pull request by cmelchior**
This adds support for fully automating 10.x releases by using tags.

After this is merged the only thing needed to publish a release is to tag a commit and then build the branch. I don't believe CI runs automatically when just pushing a tag, so I suspect this must be triggered manually (needs to be tested).

I tested as many parts of this script as I could but the entire thing hasn't run yet on CI so I suspect there are still some bugs around making it run correctly on CI with credentials.

Once reviewed I plan to use this to release BETA.7 as validation.

### CODE REVIEW

**Review by rorbech — APPROVED** [review]
Nice. 

Heavy mix of shell, Gradle and Jenskins stuff, but do not have the right separation rule from the top of my head.

File: Jenkinsfile [hunk 1]
```diff
@@ -270,6 +297,28 @@ def runBuild(abiFilter, instrumentationTestTarget) {
   }
 }
 
+def runRelease() {
+  stage('Publish Release') {
+    withCredentials([[ $class: 'StringBinding', credentialsId: 'slack-webhook-java-ci-channel', variable: 'SLACK_URL_CI']]) {
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply Jenkinsfile:302]: Ah, didn't know that was possible. Yes, that would be much more readable 👍

**Review by edualonso — APPROVED** [review]
Looks good to me 👍

**Comment by cmelchior** [comment]
Merging in preparation of BETA.7 release

**Pull request merged**